CA2241: Added try_determine_additional_string_formatting_methods_auto… (#5391)
MicrosoftDocs/visualstudio-docs: docs/code-quality/ca2241.md

- ProvideCorrectArgumentsToFormattingMethods
- CA2241

> ms.assetid: 83639bc4-4c91-4a07-a40e-dc5e49a84494

Reviewer: ```suggestion
ms.assetid: 83639bc4-4c91-4a07-a40e-dc5e49a84494
author: mikejo5000
```

> author: mikejo5000

Reviewer 2: Why was this deleted? Should this change be reverted?

Reviewer: Yes, not sure why that was deleted.

Reviewer: Added

You can configure this option for just this rule, for all rules, or for all rules in this category (Usage). For more information, see [Configure FxCop analyzers](configure-fxcop-analyzers.md).

> Instead of a specifying a long list of additional string formatting methods, you can configure the analyzer to try and determine this automatically. By default this is disabled.

Reviewer: ```suggestion
Instead of a specifying a long list of additional string formatting methods, you can configure the analyzer to try and determine this automatically. By default, this option is disabled.
```

> Instead of a specifying a long list of additional string formatting methods, you can configure the analyzer to try and determine this automatically. By default this is disabled.

Reviewer 2: ```suggestion
Instead of specifying an explicit list of additional string formatting methods, you can configure the analyzer to try and determine this automatically. By default, this option is disabled.
```

Reviewer: Merged this

> If enabled then any method that has a 'string format' parameter followed by a 'params object[]' parameter is considered a string formatting method.

Reviewer: ```suggestion
If enabled, then any method that has a `string format` parameter followed by a `params object[]` parameter is considered a string formatting method.
```

Reviewer: inline code formatting for `string format` I believe. Is that correct?

You can configure this option for just this rule, for all rules, or for all rules in this category (Usage). For more information, see [Configure FxCop analyzers](configure-fxcop-analyzers.md).

> Instead of a specifying a long list of additional string formatting methods, you can configure the analyzer to try and determine this automatically. By default, this option is disabled.

Reviewer: ```suggestion
Alternatively, instead of specifying an explicit list of additional string formatting methods, you can configure the analyzer to try and determine this automatically. By default, this option is disabled.
```
